### ISSUE TITLE
BoolBinopNode doesn't reuse temps

### ISSUE THREAD

**Issue by robertwb** [cython/cython#706]
Blocked by http://trac.cython.org/ticket/77.

Before, `BoolBinopNode` was pretty smart about only using one temp, and reuse it for its children. In the transition phase with old and new temp `ExprNodes` mixed as children, this is not possible without serious effort. Once http://trac.cython.org/ticket/77 is fixed, one could have a look at putting things back in place so that the extra C assignments aren't needed.

It is a pretty minor issue though, as no INCREF/DECREFs are involved, only the number of declared variables.

It is entirely possible to WONTFIX this ticket, but if so, please go over and clean out the comments left there now.

Migrated from http://trac.cython.org/ticket/147
